packages/lib/hooks/useTypedQuery.ts

```diff
@@ -0,0 +1,99 @@
+import { useRouter } from "next/router";
+import { useCallback, useMemo } from "react";
+import { z } from "zod";
+
+// Take array as a string and return zod array
+export const queryNumberArray = z
+  .string()
+  .or(z.number())
+  .or(z.array(z.number()))
+  .transform((a) => {
+    if (typeof a === "string") return a.split(",").map((a) => Number(a));
+    if (Array.isArray(a)) return a;
+    return [a];
+  });
```

Comment:
Support number arrays as a final "type" but input can be string so we can parse it into a number array 

query params come in as strings before being parsed